### PR TITLE
Link hints to tasks via ID

### LINKED ISSUE #6105: Hints should be matched by index, not by title
Currently, the code links a task to its hints via its title. This should be done via its index.

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes https://github.com/exercism/exercism/issues/6105﻿

### CODE REVIEW

File: app/serializers/serialize_exercise_assignment.rb [hunk 1]
```diff
@@ -87,9 +88,17 @@ def hints_doc
...
+    # "## General" becomes "General"
     header.to_plaintext.strip.gsub(/^(^\d+)\.\s*(.*)/, '\2')
   end
+
+  def parse_task_id(header)
+    # Parse the task's ID, which is the starting number.
+    # "## 2. Do stuff" becomes 2
+    # "## General" becomes 0
+    header.to_plaintext.strip.gsub(/^(^\d+)\.\s*(.*)/, '\1').to_i
+  end
```
Comment (ErikSchierboom) [inline app/serializers/serialize_exercise_assignment.rb:103]: I've opted for some duplication, but I could also be converting these into a single method returning a hash 🤷

File: app/serializers/serialize_exercise_assignment.rb [hunk 2]
```diff
@@ -32,7 +32,7 @@ def overview
   end
 
   def general_hints
-    hints["general"].to_a
+    hints[0].to_a
```
Comment (iHiD) [inline app/serializers/serialize_exercise_assignment.rb:35]: Is `general` always first? Is that in the spec?

File: app/serializers/serialize_exercise_assignment.rb (hunk 1 above)
Comment (iHiD) [reply app/serializers/serialize_exercise_assignment.rb:103]: I'm fine with either, although you **could** use the funky Ruby 3 hash-key-pattern matching stuff here probably.

File: app/serializers/serialize_exercise_assignment.rb (hunk 2 above)
Comment (ErikSchierboom) [reply app/serializers/serialize_exercise_assignment.rb:35]: No, but the first hint starts at `1` and if there is no number there (meaning: "General") the `to_i` call resolves to `0` :)

Comment (iHiD) [reply app/serializers/serialize_exercise_assignment.rb:35]: #rubyskillz

**Review by iHiD — APPROVED** [review]
I'll just trust you on this one!

**Comment by ErikSchierboom** [comment]
Well, the tests all pass and they look to be quite extensive.

**Pull request merged**